src/ansys_sphinx_theme/__init__.py

```diff
@@ -422,6 +422,49 @@ def replace_html_tag(app, exception):
             file.write(modified_content)
 
 
+def configure_theme_logo(app: Sphinx):
+    """
+    Configure the theme logo based on the theme options.
+
+    Parameters
+    ----------
+    app : sphinx.application.Sphinx
+        Application instance for rendering the documentation.
+    """
+    # Define logo configurations
+    pyansys_logo = {
+        "image_dark": pyansys_logo_dark_mode,
+        "image_light": pyansys_logo_light_mode,
+    }
+
+    ansys_logo = {
+        "image_dark": ansys_logo_dark_mode,
+        "image_light": ansys_logo_light_mode,
+    }
+    theme_options = app.config.html_theme_options
+
+    if not theme_options.get("logo"):
+        theme_options["logo"] = pyansys_logo
+
+    if isinstance(theme_options.get("logo"), str):
+        if theme_options.get("logo") == "ansys":
+            theme_options["logo"] = ansys_logo
+
+        elif theme_options.get("logo") == "pyansys":
+            theme_options["logo"] = pyansys_logo
+
+        elif theme_options.get("logo") == "no_logo":
+            theme_options["logo"] = None
+
+        else:
+            raise ValueError(
+                f"Invalid logo option: {theme_options.get('logo')}, The logo option must be either 'ansys', 'pyansys', or 'no_logo'"  # noqa: E501
+            )
```

Comment:
Trick: negate the condition to reduce indentation. Also, store the logo value to avoid checking for its value:

```suggestion
    if not isinstance(theme_options.get("logo"), str):
        raise ValueError(
            f"Invalid logo option: {theme_options.get('logo')}, The logo option must be either 'ansys', 'pyansys', or 'no_logo'"  # noqa: E501
            )
      
    logo = theme_options.get("logo")
      
    if logo == "ansys":
        theme_options["logo"] = ansys_logo
    elif logo == "pyansys":
        theme_options["logo"] = pyansys_logo
    elif theme_options.get("logo") == "no_logo":
        theme_options["logo"] = None
```